aws/data_source_aws_kms_secret.go

```diff
@@ -13,7 +13,8 @@ import (
 
 func dataSourceAwsKmsSecret() *schema.Resource {
 	return &schema.Resource{
-		Read: dataSourceAwsKmsSecretRead,
+		DeprecationMessage: "This data source will change or be removed in Terraform AWS provider version 2.0. Please see migration information available in: https://www.terraform.io/docs/providers/aws/d/kms_secrets.html",
```

Comment:
I don't know if we have precedent for provider-specific "upgrade guides" on the website, but I feel like it'd be better to gather that information on a version-specific upgrade page like we do for Core, since that way that information is obviously attached to a particular version of the provider and we won't get into a confusing situation where in future this page might no longer contain the migration information this message refers to but older builds of the provider are still out there in the wild.

If there isn't already a precedent for this then I guess we'll need to come up with a convention. My first idea would be a doc URL like `/docs/providers/aws/upgrade-guides/1-28.html`, which mimics our core-oriented upgrade guide URL structure `/upgrade-guides/0-12.html`, but I'm not wedded to that.